src/resolving-redirect-request-handler.ts

```diff
@@ -42,4 +47,27 @@ export default class ResolvingRedirectRequestHandler extends RedirectRequestHand
       response,
     };
   }
+
+  protected static purgeStorage(): void {
+    // We have discovered in practice that @openid/appauth fails to properly clean up its
+    // storage entries in some cases, namely anytime there is an error in the OAuth flow. We
+    // add this hack to manually garbage collect all outstanding storage entries that the
+    // upstream library may have left behind. Note that this fix violates modularity in two
+    // relevant ways:
+    // 1. The abstraction of StorageBackends should be respected by accessing "this.storageBackend",
```

Comment:
@zachmullen I rewrote reason 1 here to better express the real architectural pitfall of this approach, but I don't think there's a better way.